### DISCUSSION TITLE
How to add intelephense license key ?

### DISCUSSION THREAD

**Discussion by saifulapm** [helix-editor/helix#6272] [Q&A] [answered]
Hi, As intelephense has premium feature like code action, rename etc. How we can add license to intelephense .. 

Thank you

**Comment by JeftavanderHorst** [comment, marked as answer]
This is documented in the [Intelephense FAQ](https://intelephense.com/faq.html), which doesn't appear to be linked to from anywhere on the website but is mentioned in the email containing the key. Specifically, you can create a file `$HOME/intelephense/licence.txt` that contains your license key.

**Reply by saifulapm** [reply]
I have created file `$HOME/intelephense/licence.txt` then added license. But code action not working still.

**Reply by JeftavanderHorst** [reply]
Code actions also don't work for me, but other premium features like rename do. Does rename work for you?

**Comment by erasin** [comment]
create licence file like `/you-home-path/.config/php/intelephenser_licence.text`.

https://github.com/helix-editor/helix/blob/master/book/src/languages.md#languagestoml-files

create a `languages.toml` file located in your config directory:

`~/.config/helix/languages.toml`

```
[[language]]
name = "php"

[language.config]
licenceKey = "/you-home-path/.config/php/intelephenser_licence.txt"
```

The file path must be a full path. editor configuration does not support environment variables such as $HOME.

**Reply by saifulapm** [reply]
Hi.. This method not working. @JeftavanderHorst  method working fine.

**Reply by erasin** [reply]
working is good. It may be the default path. Mine is a custom path for myself.

I've been using it for a year.

https://github.com/erasin/helix-config/blob/main/languages.toml
